### PR TITLE
Automatic release workflows

### PR DESCRIPTION
**Pull request by DannyvanderJagt**
# Problem
We need to manual make a release or with a buggy script. It takes takes too much time and could be easily automated.

# Solution
Introducing a manually triggered `pre-release` workflow which handles the base32 automatically and updates all necessary files with new release version & base32 accordingly. The `publish` workflow is updated with pre-publish & post-publish checks & updates demo app automatically. 

- Support for `full, rc, alpha, beta, next, patch` releases
- Support for simultaneous releases
- Automatically validates against NPM & AWS to prevent overwriting
- Auto-publish release notes
- Automatically creates release branch & PR
- Automatically creates `master` to `development` PR for full releases

#### Base32 mapping now centralised on AWS 
A generated `base32map.json` is stored on aws & used to determine the base32 for a release instead of local `BASE_32_VERSION_MAPPING.md`

> **Related PRs:** Replace Surge with AWS https://github.com/onfido/onfido-sdk-ui/pull/1840 (waiting for Devops)

### Other changes:
- Node upgraded to v16 for all workflows
- Update `.gitignore` to fully exclude `dist` & `lib`
- Update `package.json` & `.npmignore` to only upload `dist` & `lib` excluding `reports` & `previewer`
- Fix Changelog typos
- Remove `node release/release.js` script

## TODO:
- [x] Wait for DevOps to update AWS token
- [ ] Test on prod
- [ ] Create PR for merge master to development automatically

## Before merge:
Reset/remove files:
- [x] `prepare_release.json`
- [x] `release.json`
- [x] `webpack.config.babel.js`
- [x] `release/githubActions/workflows.config`
- [x] `demo/fiddle/demo.details`

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [ ] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: release/updateChangeLog.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      ].join('')
+    )
+    .join('')
+
+  const links = data.versions
+    .map((i, index) => {
+      const isNextVersion = index === 0
+      const isLastEntry = index === data.versions.length - 1
+
+      const previous = isLastEntry ? '0.4.0' : data.versions[index + 1]?.version
```
Comment (stefaniacardenas) [inline release/updateChangeLog.js:81]: What's this hard coded value here?

File: .github/workflows/prepare_release.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "inputs": {
+        "release_version": "12.0.0",
+        "is_lts": false
+    }
+}
```
[the remark this answers is not in the document]
Comment (DannyvanderJagt) [reply .github/workflows/prepare_release.json:6]: I'm using a tool called `act` (https://github.com/nektos/act) to run github action/ci locally. The tool uses those files as input, I will remove them before merging once everything is completed.

File: package.json [hunk 3]
```diff
@@ -53,11 +53,6 @@
     "identity",
     "verification"
   ],
-  "files": [
-    "dist",
-    "lib",
-    "src"
-  ],
```
[the remark this answers is not in the document]
Comment (DannyvanderJagt) [reply package.json:60]: I've updated to `.npmignore` that will do this.

File: release/updateChangeLog.js (hunk 1 above)
Comment (DannyvanderJagt) [reply release/updateChangeLog.js:81]: The earliest version mentioned in our CHANGELOG.md is `0.5.0` and our earliest release tag on github is `0.4.0`. In order to construct the first url on the bottom of our change log (for `0.5.0`) we need `0.4.0` but as it isn't in the changelog, I opted for a simple hard coded value as it won't change anymore.

Comment (stefaniacardenas) [reply release/updateChangeLog.js:81]: We can assign it to a variable e.g. `const oldestVersion = '0.4.0'`

File: .github/workflows/prepare_release.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (stefaniacardenas) [reply .github/workflows/prepare_release.json:6]: Ah cool, I was using act too. It's a good tool :)

**Review by stefaniacardenas — APPROVED** [review]
Looks good to me! 🚀  Just remember to remove unnecessary files before merge 🙏

**Pull request merged**